File: src/dynamicprogramming/MaxProfit.java
package dynamicprogramming;

public class MaxProfit {
    public int maxProfit(int[] prices) {
        int account = 0;
        int m = 0, t = 0;
        for (int i = 0; i < prices.length; i++) {
            if (m != 0 && prices[i] < prices[i + 1]) {
                m = prices[i];
            }

            if (prices[i] > prices[i + 1] && m != 0) {
                t = prices[i];
                account += t - m;
            }
        }
        return 0;
    }
}

File: LeetcodeKt/src/array/MaxProfit.kt
package array
/*
#122. 买卖股票的最佳时机 II
https://leetcode-cn.com/problems/best-time-to-buy-and-sell-stock-ii/
 */
class MaxProfit {
    fun maxProfit(prices: IntArray): Int {
        var res = 0
        for (i in 0 until prices.size - 1) {
            if (prices[i] < prices[i + 1]){
                res += prices[i+1] - prices[i]
            }
        }
        return res
    }
}